lesson_9/recursion.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * @param {number} number 
+ * @param {number} pow 
+ * @returns {number} number
+ */
+
+function numberToPow(number, pow) {
+  if (pow >= 0) {
+    const result = pow === 0 ? 1 : number * numberToPow(number, --pow);
+    return result;
+  } else {
+    if (pow === -1) {
+      const result = 1.0 / number;
+      return result;
+    } else {
+      const result = 1.0 / (number * numberToPow(number, (-1) * (pow + 1)));
+      return result;
```

Comment:
as far as `result` don't extend you function with any logic you can avoid redundant variables in this case
```suggestion
    return pow === 0 ? 1 : number * numberToPow(number, --pow);
  } else {
    if (pow === -1) {
      return 1.0 / number;
    } else {
      return 1.0 / (number * numberToPow(number, (-1) * (pow + 1)));;
```